lib/ClangImporter/ClangImporter.cpp

```diff
@@ -358,6 +358,39 @@ getNormalInvocationArguments(std::vector<std::string> &invocationArgStrs,
       // Just use the most feature-rich C language mode.
       "-x", "c", "-std=gnu11",
     });
+
+    // The module map used for Glibc depends on the target we're compiling for,
+    // and is not included in the resource directory with the other implicit
+    // module maps. It's at {freebsd|linux}/{arch}/glibc/module.map.
+    llvm::SmallString<128> GlibcModuleMapPath;
+    if (!importerOpts.OverrideResourceDir.empty()) {
+      GlibcModuleMapPath =
+        llvm::SmallString<128>(importerOpts.OverrideResourceDir);
+    } else if (!searchPathOpts.RuntimeResourcePath.empty()) {
+      GlibcModuleMapPath =
+        llvm::SmallString<128>(searchPathOpts.RuntimeResourcePath);
+    }
+    if (!GlibcModuleMapPath.empty()) {
+      if (triple.isOSFreeBSD()) {
+        llvm::sys::path::append(GlibcModuleMapPath, "freebsd");
+      } else {
+        llvm::sys::path::append(GlibcModuleMapPath, "linux");
+      }
+      llvm::sys::path::append(
+        GlibcModuleMapPath, triple.getArchName(), "glibc", "module.map");
+
+      // Only specify the module map if that file actually exists.
+      // It may not--for example in the case that
+      // `swiftc -target x86_64-unknown-linux-gnu -emit-ir` is invoked using
+      // a Swift compiler not build for Linux targets.
+      if (llvm::sys::fs::exists(GlibcModuleMapPath)) {
+        invocationArgStrs.insert(invocationArgStrs.end(), {
+          (Twine("-fmodule-map-file=") + GlibcModuleMapPath).str(),
```

Comment:
This goes through `std::string`; it would be nice™ to avoid the allocation cost by continuing to use SmallString.